plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,375 @@
+
+    l3out_object = mso_template.get_l3out_object(name=l3out, fail_module=True)
+    l3out_node_group = mso_template.get_l3out_node_group(name, l3out_object.details)
+
+    if name and l3out_node_group:
+        mso.existing = mso.previous = copy.deepcopy(l3out_node_group.details)  # Query a specific object
+    elif l3out_node_group:
+        mso.existing = l3out_node_group  # Query all objects
+
+    if state != "query":
+        node_group_policy_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}".format(l3out_object.index, l3out_node_group.index if l3out_node_group else "-")
+
+    ops = []
+
+    if state == "present":
+        l3out_node_routing_policy_object = None
+        if node_routing_policy:
+            l3out_node_routing_policy_objects = mso.query_objs(
+                generate_api_endpoint(
+                    "templates/objects", **{"type": "l3OutNodePolGroup", "tenant-id": mso_template.template_summary.get("tenantId"), "include-common": "true"}
+                )
+            )
+            l3out_node_routing_policy_object = mso_template.get_object_by_key_value_pairs(
+                "L3Out Node Routing Policy", l3out_node_routing_policy_objects, [KVPair("name", node_routing_policy)], True
+            )
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+            if proposed_payload.get("bfdMultiHop", {}).get("key", {}).get("ref"):
+                proposed_payload["bfdMultiHop"]["key"].pop("ref", None)
+                mso.existing["bfdMultiHop"]["key"].pop("ref", None)
+
+            if description is not None and mso.existing.get("description") != description:
+                ops.append(dict(op="replace", path=node_group_policy_path + "/description", value=description))
+                proposed_payload["description"] = description
+
+            if l3out_node_routing_policy_object and mso.existing.get("nodeRoutingPolicyRef") != l3out_node_routing_policy_object.details.get("uuid"):
+                ops.append(
+                    dict(op="replace", path=node_group_policy_path + "/nodeRoutingPolicyRef", value=l3out_node_routing_policy_object.details.get("uuid"))
+                )
+                proposed_payload["nodeRoutingPolicyRef"] = l3out_node_routing_policy_object.details.get("uuid")
+
+            elif node_routing_policy == "" and mso.existing.get(
+                "nodeRoutingPolicyRef"
+            ):  # Clear the node routing policy when node_routing_policy is empty string
+                ops.append(dict(op="replace", path=node_group_policy_path + "/nodeRoutingPolicyRef", value=node_routing_policy))
+                proposed_payload["nodeRoutingPolicyRef"] = node_routing_policy
+
+            if bfd:
+                if bfd.get("state") == "disabled" and proposed_payload.get("bfdMultiHop"):
+                    proposed_payload.pop("bfdMultiHop")
+                    ops.append(dict(op="remove", path=node_group_policy_path + "/bfdMultiHop"))
+                else:
```

Comment:
I think this else case could run with state=disabled, when the configuration is not in schema. Would expect elif case here